### PR TITLE
Latent MOE & Repeated MTP support for NemotronH; fix KV cache quant export

### PR DESCRIPTION
**Pull request by jenchen13**
## What does this PR do?

**Type of change:** New feature and bug fix

**Overview:** 

Support Latent MOE and Repeated MTP for NemotronH models
- Enable latent MOE modules during megatron import/export
- Fix KV cache quantization export: remove old `qkv_layer.output_quantizer` export & replace with proper `k/v_bmm_quantizer` logic
- Improvements to EP amax sync
- Support repeated MTP import/export for NemotronH models (only BF16 export for MTP for now)

## Usage

```python
# Add a code snippet demonstrating how to use this
```

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes/No 
- **Did you write any new necessary tests?**: Yes/No
- **Did you add or update any necessary documentation?**: Yes/No
- **Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?**: Yes/No 

## Release Notes

* **New Features**
  * Added support for grouped MLP and self-attention scaling operations in model export workflows
  * Enhanced model parallel training capabilities with improved component mapping
  * Expanded quantization configuration handling with dynamic module exclusion across distributed ranks
  * Improved support for additional transformer engine components

* **Refactor**
  * Reorganized internal export and import logic for improved maintainability and specialist model architecture support

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: modelopt/torch/quantization/model_calib.py [hunk 1]
```diff
@@ -81,6 +81,11 @@ def max_calibrate(model: nn.Module, forward_loop: ForwardLoop | None = None, dis
         forward_loop(model)
     finish_stats_collection(model)
 
+    # Step 1: Sync amax across local experts in a SequentialMLP
+    for name, module in model.named_modules():
+        if hasattr(module, "sync_moe_local_experts_amax"):
+            module.sync_moe_local_experts_amax()
```
Comment (ChenhanYu) [inline modelopt/torch/quantization/model_calib.py:118]: This function says to sync across EP and ETP but I didn't see any collective communication happen in the function. The description needs to be changed I think.

Comment (jenchen13) [reply modelopt/torch/quantization/model_calib.py:118]: I can change the name of the method. What it's actually doing is just looping through the local experts and assigning the same amax

**Comment by cjluo-nv** [comment]
I have concerns on "Remove KV quantization scale clamping to 1" without accuracy data support. This also deviates with the HF ptq code path. I think we need to have a discussion about the impact.

File: modelopt/torch/quantization/model_calib.py (hunk 1 above)
Comment (realAsma) [reply modelopt/torch/quantization/model_calib.py:118]: it is syncing local experts for sequential moe.

Comment (realAsma) [reply modelopt/torch/quantization/model_calib.py:118]: the name does not say or indicate it is collective comm.
